EIPS/eip-draft_transaction_data_opcodes.md

```diff
@@ -0,0 +1,106 @@
+
+### ENTRYPOINT (`0x4a`)
+
+The `ENTRYPOINT` instruction uses 0 stack arguments and pushes the original `to` member of the transaction onto the stack. The address yielded by the instruction is a 160-bit value padded to 256-bits. The operation costs `G_base` to execute, similarly to `ORIGIN` (`0x32`).
+
+### Node Specification
+
+All instructions are meant to provide access to previously inaccessible data that did not exist in the execution context of the EVM. As a result, the data needs to be provided to the execution environments of all contract calls, similarly to how `ORIGIN` data is available as such. A reference implementation will be provided for `go-ethereum` (Golang) that implements the EIP at both the EVM level as well as the node level.
+
+### Compiler Specification
+
+High-level programming languages that compile to EVM bytecode should provide support for these new instructions from a high-level perspective by introducing new members to the transactional context they already support for the `ORIGIN` instruction.
+
+## Rationale
+
+### Naming Conventions
+
+The `ORIGIN`-prefixed instructions attempted to conform to the existing naming convention of `CALL`-prefixed instructions given the existence of the `ORIGIN` instruction which is equivalent to the `CALLER` (`0x33`) instruction but on the original transaction's context.
+
+The `ENTRYPOINT` instruction came to be by defining a sensible name that immediately and clearly depicts what it is meant to achieve by signaling the first interaction with the blockchain, i.e. the entry-point.
+
+### Instruction Address Space
+
+The instruction address space of the `0x30-0x3f` has been exhausted by calls that already provide information about the execution context of a call so a new range had to be identified that is suitable for the purposes of the EIP.
+
+Given that the [EIP-1344](https://eips.ethereum.org/EIPS/eip-1344) `CHAINID` opcode was included at `0x46`, it made sense to include additional transaction-related data beyond it since the Chain ID is also included in transaction payloads apart from the blocks themselves, rendering the `0x46-0x4f` address space reserved for more transaction-related data that may be necessary in the future, such as the EOA's nonce.
+
+### Gas Costs
+
+The gas costs necessary for a particular instruction must weight the computational expense it brings to the system to assess a particular number. In the case of the instructions proposed by this EIP, this assessment was relatively straightforward given that identical instructions have already been introduced to the EVM and the instructions themselves would operate in an equivalent manner.
```

Comment:
Recommend simplifying this to just (fill in X, Y, Z):
```suggestion
These opcodes do essentially the same thing as opcodes X, Y, Z and thus share cast costs with those opcodes.
```

Reply:
Done, kindly review.